tree/ntuple/v7/inc/ROOT/RNTupleMerger.hxx

```diff
@@ -97,11 +98,12 @@ class RNTupleMerger final {
                             std::span<RColumnMergeInfo> extraDstColumns, RNTupleMergeData &mergeData);
 
 public:
-   RNTupleMerger();
+   /// Creates a RNTupleMerger with the given destination.
+   /// `destination` must outlive the RNTupleMerger.
+   explicit RNTupleMerger(RPageSink &destination);
```

Comment:
Wouldn't passing a `std::shared_ptr` be a better indication of the require lifetime?

Reply:
I don't think so, a) because it's clear that the merger should have a strictly smaller lifetime than the Sink it's operating on (whereas the `shared_ptr` would blur this concept) and, more importantly b) our API always returns PageSinks/Sources as `unique_ptr`.

Reply:
Hm, but how would the user get back ownership if they need it when they are done with it? If we add a `RetrieveSink()` method (or whatever) we leave the Merger in an invalid state..

Reply:
I don't think the user should ever get it back; as we discussed the merger will soon own the model, so I'm not sure the user could even do something useful with the sink except merging in more files

Reply:
Makes sense

Reply:
Using the `unique_ptr` solves the problem.

> because it's clear that the merger should have a strictly smaller lifetime than the Sink

Yes for a use case where the user still need access to the object (and thus unique_ptr is not an option), capturing the reference is 'silently' sharing the ownership/usage of the object and will lead to user error.  We are trying to move away from implicitness.  i.e. `should have a strictly smaller lifetime ` might be true but can not be enforced nor checked by the compiler (nor simple static analysis tools).

Reply:
@pcanal so we have an agreement on the unique_ptr right? Can this thread be resolved?